app/kumactl/cmd/apply/apply_test.go

```diff
@@ -25,6 +26,7 @@ var _ = Describe("kumactl apply", func() {
 	var rootCtx *kumactl_cmd.RootContext
 	var rootCmd *cobra.Command
 	var store core_store.ResourceStore
+	var port string = ":3000"
```

Comment:
Pick a random port free here. It's always good practice to avoid hardcoding port in tests. This way if you have an app in your system on port 3000, the test will not be affected by this.

Also, use `httptest` package. Look for example in
kuma/app/kuma-dp/pkg/dataplane/envoy/remote_bootstrap_test.go